src/components/searchBar.tsx

```diff
@@ -0,0 +1,57 @@
+import { CharacterResultsStatesByName } from '../interfaces';
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import GetCharacterByName from '../rickAndMortyApi/GetCharacterByName';
+import './SearchBar.css';
+
+const SearchBar = ({ updateCartList, loading }: CharacterResultsStatesByName) => {
+  const searchRef = useRef<string>('');
+  const [searchState, setSearchState] = useState(localStorage.getItem('searchValue') || '');
+  const inputHandler = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value;
+    setSearchState(inputValue);
+    localStorage.setItem('searchValue', inputValue);
+  };
+
+  useEffect(() => {
+    return () => {
+      localStorage.setItem('searchValue', searchRef.current || '');
+    };
+  }, []);
+
+  useEffect(() => {
+    searchRef.current = searchState;
+  }, [searchState]);
+
+  const clearHandleClick = () => {
+    setSearchState('');
+    localStorage.removeItem('searchValue');
+  };
+
+  const sendHandleClick = async () => {
+    loading(true);
+    GetCharacterByName(searchState)
+      .then((characterByName) => {
+        updateCartList(characterByName.results);
+        loading(false);
+      })
+      .catch(() => {
+        updateCartList([]);
+        loading(false);
+      });
```

Comment:
It's better to use finally like this:

GetCharacterByName(searchState)
      .then((characterByName) => {
        updateCartList(characterByName.results);
        // loading(false);
      })
      .catch(() => {
        updateCartList([]);
        // loading(false);
      })
      .finally(() => {
        loading(false);
      });